### PR TITLE
LPD-30488 Opensearch 2 Bump to 2.16

### PR DESCRIPTION
**Pull request by joshuacords**
https://liferay.atlassian.net/browse/LPD-30490
https://liferay.atlassian.net/browse/LPD-32382

Continued from https://github.com/liferay-search/liferay-portal/pull/1457

"Missing required property 'GetSnapshotResponse.total'" exception is not a blocker for opensearch release.

**Author:** @joshuacords

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search-opensearch2

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search-opensearch2

**Comment by joshuacords** [comment]
ci:report:79319110

**Comment by joshuacords** [comment]
ci:report:79220484

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by peerkar** [comment]
@joshuacords Generally, I don't think those tests expectations should be created based on the bugs of OS Java client, but based on the server messages. I think the client should be transparent.  Can't we just simply set these tests to ignored and add there a comment referring to the LPD?

**Comment by joshuacords** [comment]
Hi @peerkar,

The expectations being an exception is thrown was based on a conversation Bryan and I had - the reasoning is that giving them just @ignore wouldn't guarantee that they would be reverted once the issue is addressed, where the Expected Exception would. I'm up for doing either though.

cc: @BryanEngler

File: modules/apps/portal-search-opensearch2/portal-search-opensearch2-impl/src/main/java/com/liferay/portal/search/opensearch2/internal/search/engine/adapter/snapshot/OpenSearchSnapshotRequestExecutor.java [hunk 1]
```diff
@@ -66,7 +66,8 @@ public GetSnapshotRepositoriesResponse executeSnapshotRequest(
 	public GetSnapshotsResponse executeSnapshotRequest(
 		GetSnapshotsRequest getSnapshotsRequest) {
 
-		return getSnapshotsRequestExecutor.execute(getSnapshotsRequest);
+		throw new UnsupportedOperationException(
+			"Opensearch's java-client does not support GetSnapshotRequest");
```
Comment (joshuacords) [inline modules/apps/portal-search-opensearch2/portal-search-opensearch2-impl/src/main/java/com/liferay/portal/search/opensearch2/internal/search/engine/adapter/snapshot/OpenSearchSnapshotRequestExecutor.java:70]: make "java client"

Comment (joshuacords) [reply modules/apps/portal-search-opensearch2/portal-search-opensearch2-impl/src/main/java/com/liferay/portal/search/opensearch2/internal/search/engine/adapter/snapshot/OpenSearchSnapshotRequestExecutor.java:70]: Actually just remove commit

**Comment by BryanEngler** [comment]
@peerkar i didnt fully understand what you meant by "but based on the server messages"?

**Comment by peerkar** [comment]
>  i didnt fully understand what you meant by "but based on the server messages"?

As these are client tests, generally spoken, I think we should test against explicit exceptions thrown by the server. If we test exceptions thrown from the middle, there's a risk the actual server side causes stay in the shadows, might even change by time time, making the test even not necessarily that purposeful anymore. That kind of tests do not necessarily test the contract between the server and the client any more, which I think should be the main purpose.

An example what I mean is [here](https://github.com/liferay/liferay-portal/blob/2aa20937beaf1415ddacba4447d5aee553580e0d/modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/logging/ElasticsearchIndexWriterExceptionsTest.java#L203-L222). Our own "Update failed" or "Bulk update failed" messages can be thrown for various server side reasons, which might even change by the time. The coverage of tests like this is for me thus somewhat vague. 

> i think we can keep the expected exceptions. if we ignore, then we have to 1) remember to unignore 2) continue repeating the process each time we do a bump if the bug remains unfixed. if we just expect the exceptions, then if/when the bug is fixed later on the tests will automatically fail and we will know to update them at that point.

Makes sense :+1:

**Pull request closed without merging**